design-documents/graph-ql/coverage/catalog/configurable-options-selection.graphqls

```diff
@@ -0,0 +1,20 @@
+type ConfigurableProduct {
+    configurable_options_selection_metadata(selectedConfigurableOptionValues: [ID!]): ConfigurableOptionsSelectionMetadata @doc(description: "Metadata for the specified configurable options selection")
+}
+
+type ConfigurableOptionsSelectionMetadata @doc(description: "Metadata corresponding to the configurable options selection.")
+{
+    configurable_options_available_for_selection: [ConfigurableOptionAvailableForSelection!] @doc(description: "Configurable options available for further selection based on current selection.")
```

Comment:
I _think_ we can make `configurable_options_available_for_selection` a bit more concise without losing meaning.

Because this field lives inside the `ConfigurableOptionsSelectionMetadata` type, and is being plucked from a field named `configurable_options_selection_metadata `, seems like the `configurable_options` prefix is a bit redundant.

```graphql
{
   configurable_options_selection_metadata {
      configurable_options_available_for_selection {
         # Can drop the prefix on the field without losing meaning
      }
   }
}
```

Reply:
What about something like this?

```graphql
{
   ...on ConfigurableProduct {
      options_selections {
         available
      }
   }
}
```

Not married to the specific names, but I think the query is a bit more self-describing on first read.

Another small benefit is that the client-side doesn't have to keep repeating `configurable_options_something_something` everywhere the object is passed around (or alias field usage every time).

Reply:
Another idea

```graphql
query GetUpdatedOptionsValues($sku: String!, $selectedOptionIDs: [ID!]) {
  products(filter: {sku: {eq: $sku}}) {
	...on ConfigurableProduct {
       availableOptions(selectedOptionIDs: $selectedOptionIDs) {
         optionGroups {
           attribute_code
           available_value_ids
         }
       }
   }
  }
}
```

Reply:
I believe having "selection metadata" is very important. Because it not only contains available options, but selection metadata like variant, price range, low stock status etc.

"configurable options" prefix is added for consistency with other product type specific fields like `configurable_options`. 

I'm okay with renaming `configurable_options_available_for_selection` to `options_available_for_selection` since it is located under `configurable_options_selection_metadata`. Reducing it further will hurt the meaning I think.

Reply:
> "configurable options" prefix is added for consistency with other product type specific fields like configurable_options.

That's reasonable. The repetition is a bit annoying for a client, but the trade-off is that we're less likely to have field name collisions down the line, which we've already learned is a real problem 😅

> I'm okay with renaming configurable_options_available_for_selection to options_available_for_selection since it is located under configurable_options_selection_metadata. Reducing it further will hurt the meaning I think.

Cool! Let's go with that.

Reply:
Done